lib/ddtrace/context.rb

```diff
@@ -133,12 +133,14 @@ def sampled?
     # can be re-used immediately.
     #
     # This operation is thread-safe.
-    def get
+    def get(priority_sampling = false)
       @mutex.synchronize do
         return nil, nil unless check_finished_spans
 
         trace = @trace
         sampled = @sampled
+        attach_sampling_priority if sampled && priority_sampling
```

Comment:
I'm not sure why you need to test `priority_sampling` here, at all. This "is priority sampling enabled" is useful when creating the tracer, so that we know wether `start_span` should add a priority sampling arg, but then, once it's been set, it's been set, there's, I think, no point to re-test it here again. What is the use case for this?

Reply:
humn, I thought we had to check it here so we can avoid a scenario like this:

*In my custom instrumentation, I have a code line like this:*

```rb
span.context.sampling_priority = 3
```

*But at some point, I want to disable the priority sampling altogether:*
```rb
Datadog.tracer.configure(priority_sampling: false)
```
If we don't check for the priority sampling flag before flushing the trace, that line setting a `sampling_priority` will still affect the sampling upstream, right?

I don't understand what happens outside the client, so I might be wrong. But if that's not the case, I would also prefer to remove this check. We can take a stance as "if you want to disable priority_sampling altogether you shouldn't set it."

BTW, I thought this was the scenario you were trying to prevent in that `delete` call inside the `HTTPPropagator.inject!` method.